fem/bilininteg.hpp

```diff
@@ -2999,6 +3029,17 @@ class VectorInnerProductInterpolator : public DiscreteInterpolator
    VectorCoefficient *VQ;
 };
 
-}
 
+
+// PA Diffusion Assemble 2D kernel
+template<const int T_SDIM>
+void PADiffusionSetup2D(const int Q1D,
```

Comment:
I'm really not a fan of this method being in `bilininteg.hpp`, I also think that using templates is not useful here since you specialize each implementation and you don't have a generic implementation.

Reply:
This was a template before this PR, but I agree it could be two separate functions named `PADiffusionSetup2D` and `PADiffusionSetup2Din3D`, for example. The declaration is in the header because it is being used in diffusion and H(curl) files. In bilininteg_vectorfe.cpp there is just a forward declaration of `PADiffusionSetup3D`. I'm not sure if that would work for the 2D template function, but I could change the template function to 2 functions with different names and put their declarations in bilininteg_vectorfe.cpp. Does that sound good?